cli/checkpoint/lua/cat.lua

```diff
@@ -0,0 +1,219 @@
+-- This is a script that print into stdout the contents of .snap/.xlog files.
+-- The files passes throught "TT_CLI_CAT_FILES".
+-- The --show-system flag passes throught "TT_CLI_CAT_SHOW_SYS".
+-- The --space flags passes throught "TT_CLI_CAT_SPACES".
+-- The --from flag passes throught "TT_CLI_CAT_FROM".
+-- The --to flag passes throught "TT_CLI_CAT_TO".
+-- The --replica flags passes throught "TT_CLI_CAT_REPLICAS".
+-- The --format flags passes throught "TT_CLI_CAT_FORMAT".
+
+local log = require('log')
+local xlog = require('xlog')
+local yaml  = require('yaml')
+local json = require('json')
+
+local function cat_yaml_cb(record)
+    print(yaml.encode(record):sub(1, -6))
+end
+
+local function cat_json_cb(record)
+    print(json.encode(record))
+end
+
+local function write_lua_string(string)
+    io.stdout:write("'")
+    local pos, byte = 1, string:byte(1)
+    while byte ~= nil do
+        io.stdout:write(("\\x%02x"):format(byte))
+        pos = pos + 1
+        byte = string:byte(pos)
+    end
+    io.stdout:write("'")
+end
+
+local write_lua_table = nil
+
+local function write_lua_value(value)
+    if type(value) == 'string' then
+        write_lua_string(value)
+    elseif type(value) == 'table' then
+        write_lua_table(value)
+    else
+        io.stdout:write(tostring(value))
+    end
+end
+
+local function write_lua_fieldpair(key, val)
+    io.stdout:write("[")
+    write_lua_value(key)
+    io.stdout:write("] = ")
+    write_lua_value(val)
+end
+
+write_lua_table = function(tuple)
```

Comment:
Why not `local function write_lua_table(tuple)`?

Reply:
```
local write_lua_table = nil    -- function declaration

write_lua_table = function(tuple)    -- function definition
  ...
end
```

AFAIU, without separation into declaration and definition for `write_lua_table`, functions `write_lua_fieldpair` and `write_lua_value` cannot be used. This was inherited from the original implementation

Reply:
Fixed